vcstool/commands/compare.py

```diff
@@ -1,80 +1,595 @@
 import argparse
-from cgitb import reset
-from distutils.util import execute
-from gettext import find
-from prettytable import PrettyTable
 import sys
-from unittest import result
-from vcstool import clients
-from vcstool.commands.branch import BranchCommand
-from vcstool.crawler import find_repositories
+import os
+import abc
+from typing import Any, Dict, List, Optional
+from enum import IntEnum, IntFlag, auto
+
+import prettytable as pt
 
-from vcstool.executor import ansi, execute_jobs, generate_jobs, output_repositories
+from vcstool.crawler import find_repositories
+from vcstool.executor import ansi, execute_jobs, generate_jobs
 from vcstool.commands.import_ import get_repositories
 from vcstool.streams import set_streams
+from vcstool.outputs import CompareOutput
 
 from .command import add_common_arguments
 from .command import Command
 
 
 class CompareCommand(Command):
 
-    command = 'compare'
-    help = 'Compare working copy to the repository list file'
+    command = "compare"
+    help = "Compare working copy to the repository list file"
 
-    def __init__(self, args):
-        super(CompareCommand, self).__init__(args)
+    def __init__(self, args: argparse.Namespace) -> None:
+        super().__init__(args)
+        self.progress = args.progress
+        self.workers = args.workers
+        self.debug = args.debug
 
+    @classmethod
+    def get_parser(cls) -> argparse.ArgumentParser:
+        parser = argparse.ArgumentParser(description=f"{cls.help}", prog="vcs compare")
+        group = parser.add_argument_group('"compare" command parameters')
+        group.add_argument(
+            "-i",
+            "--input",
+            type=str,
+            default="workspace.repos",
```

Comment:
I know we use _workspace.repos_ now but do you think we should not have a default and force user to specify the repos input file for scalability/modularity? We can always alias the `vcs compare` command.

Reply:
I can actually add a feature to scan the `cwd` to find _any_ `.repos` file. If only one exists, we can use that. If multiple exist, we can error and force the user to specify the file.

Reply:
Lets get input on this from a few others. My only concern is that the _src_ directory and repos file in our case is at the same level but in other ROS workspaces is within the _src_ directory. Not sure if it is better to force usage to be more explicit or remove burden on the user.

Reply:
I added the functionality to this PR, but it can be easily reverted